plugins/modules/file_interface.py

```diff
@@ -0,0 +1,465 @@
+        for key in modify_keys:
+            if file_interface_params[key] is not None and \
+                    file_interface_params[key] != file_interface_details[key]:
+                modify_dict[key] = file_interface_params[key]
+        if modify_dict:
+            return modify_dict
+        else:
+            return None
+
+    def modify_file_interface_details(self, file_interface_id,
+                                      modify_params):
+        """Perform modify operations on a file interface"""
+
+        try:
+            if not self.module.check_mode:
+                self.file_interface.modify_file_interface(
+                    file_interface_id=file_interface_id,
+                    modify_parameters=modify_params)
+            return True
+        except Exception as e:
+            msg = (f'Failed to modify the file interface instance '
+                   f'with error {str(e)}')
+            LOG.error(msg)
+            self.module.fail_json(msg=msg, **utils.failure_codes(e))
+
+
+def get_powerstore_file_interface_parameters():
+    """This method provides the parameters required for the ansible
+    nas server modules on PowerStore"""
+    return dict(
+        nas_server_name=dict(type='str'),
+        nas_server_id=dict(type='str'),
+        file_interface_id=dict(type='str'),
+        ip_address=dict(type='str'),
+        gateway=dict(type='str'),
+        prefix_length=dict(type='int'),
+        vlan_id=dict(type='int'),
+        ip_port_id=dict(type='str'),
+        is_disabled=dict(required=False, type='bool'),
+        role=dict(required=False, type='str',
+                  choices=['Production', 'Backup', 'System']),
+        is_destination_override_enabled=dict(type='bool'),
+        state=dict(default='present', type='str', choices=['present', 'absent'])
+    )
+
+
+class FileInterfaceExitHandler():
+    def handle(self, file_interface_obj, file_interface_details):
+        file_interface_obj.result["file_interface_details"] = file_interface_details
+        file_interface_obj.module.exit_json(**file_interface_obj.result)
+
+
+class FileInterfaceDeleteHandler():
+    def handle(self, file_interface_obj, file_interface_params, file_interface_details):
+        if file_interface_params['state'] == 'absent' and file_interface_details:
+            changed = file_interface_obj.delete_file_interface(file_interface_params['file_interface_id'])
+            file_interface_obj.result['changed'] = True
+            file_interface_details = {}
+
+        FileInterfaceExitHandler().handle(file_interface_obj, file_interface_details)
+
+
+class FileInterfaceModifyHandler():
+    def handle(self, file_interface_obj, file_interface_params, file_interface_details):
+        if file_interface_params['state'] == 'present' and file_interface_details:
+            modify_dict = file_interface_obj.is_modify_required(file_interface_details, file_interface_params)
+            if modify_dict:
```

Comment:
As this check already added for modify_dict in **is_modify_required** method. You can remove this check from either of the place.